geoapps/drivers/components/locations.py

```diff
@@ -0,0 +1,83 @@
+#  Copyright (c) 2021 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+from uuid import UUID
+
+import numpy as np
+from geoh5py.objects import Grid2D
+
+from geoapps.utils import rotate_xy
+
+
+class InversionLocations:
+    """ Retrieve topography data from workspace and apply transformations. """
+
+    def __init__(self, workspace, params, mesh, window):
+        self.workspace = workspace
+        self.params = params
+        self.mesh = mesh
+        self.window = window
+        self.mask = None
+        self.origin = None
+        self.angle = None
+        self.is_rotated = False
+        self.locs = None
+        self._initialize()
+
+    def _initialize(self):
+
+        if self.mesh.rotation is not None:
+            self.origin = self.mesh.rotation["origin"]
+            self.angle = -self.mesh.rotation["angle"]
+            self.is_rotated = True
+
+    @property
+    def mask(self):
+        return self._mask
+
+    @mask.setter
+    def mask(self, v):
+        if v is None:
+            self._mask = v
+            return
+        if np.all([n in [0, 1] for n in np.unique(v)]):
+            v = np.array(v, dtype=bool)
+        else:
+            msg = f"Badly formed mask array {v}"
+            raise (ValueError(msg))
+        self._mask = v
+
+    def get_locs(self, uid: UUID) -> np.ndarray:
+        """
+        Returns locations of data object centroids or vertices.
+
+        :param: uid: UUID of geoh5py object containing centroid or
+            vertex location data
+        :returns: locs: N by 3 array of x, y, z location data
+
+        """
+
+        data_object = self.workspace.get_entity(uid)[0]
+
+        if isinstance(data_object, Grid2D):
+            locs = data_object.centroids
+        else:
+            locs = data_object.vertices
```

Comment:
This might break weird if the user inputs something else than a Grid2D. Maybe add a check after for `locs is not None`?

Reply:
Fixed up.  BlockModel initializes with vertices=None as you suspected.